examples/hello-world-java/src/main/java/org/classysheet/examples/GoogleSheetsExample.java

```
@@ -3,16 +3,34 @@
 import org.classysheet.core.api.ClassysheetService;
 import org.classysheet.examples.data.ScheduleGenerator;
 import org.classysheet.examples.domain.Schedule;
+import org.classysheet.core.impl.provider.google.SheetsServiceUtil;
+import com.google.api.services.sheets.v4.Sheets;
+import com.google.api.services.sheets.v4.model.Spreadsheet;
+import com.google.api.services.sheets.v4.model.SpreadsheetProperties;
 
 public class GoogleSheetsExample {
-
     public static void main(String[] args) {
-        // Once, at build or bootstrap time:
-        ClassysheetService<Schedule> classysheetService = ClassysheetService.create(Schedule.class);
+        try {
+            Sheets sheetsService = SheetsServiceUtil.getSheetsService();
 
-        // At runtime, per workbook:
-        Schedule schedule = ScheduleGenerator.generateDemoData();
-        classysheetService.writeWorkbookToGoogle(schedule);
-    }
+            Spreadsheet newSpreadsheet = new Spreadsheet()
+                    .setProperties(new SpreadsheetProperties().setTitle("ClassySheet Example Workbook"));
+            Spreadsheet createdSpreadsheet = sheetsService.spreadsheets().create(newSpreadsheet).execute();
+            String spreadsheetId = createdSpreadsheet.getSpreadsheetId();
+            if (spreadsheetId == null) {
+                throw new RuntimeException("Error creating workbook on google sheets");
+            }
+
+            ClassysheetService<Schedule> classysheetService = ClassysheetService.<Schedule>create(Schedule.class, spreadsheetId);
+
+            Schedule schedule = ScheduleGenerator.generateDemoData();
 
+            classysheetService.writeWorkbookToGoogle(schedule);
+
+            System.out.println("data successfully written on" + spreadsheetId);
+
+        } catch (Exception e) {
+            e.printStackTrace();
```

This eats the exception. That's bad because it doesn't bring the whole request down, but doesn't do what it's expected to do. That looses trust with users and make it harder to work toward bug free applications.

It's better to catch more specifically the checked exceptions you need to catch and rethrow them as an IllegalStateException("...", e) or IllegalArgumentException("...", e)